app/src/pages/playground/PlaygroundToolCall.tsx

```diff
@@ -1,48 +1,90 @@
 import React, { useMemo } from "react";
 import { css } from "@emotion/react";
 
+import { Text } from "@arizeai/components";
+
 import {
   detectToolCallProvider,
   LlmProviderToolCall,
 } from "@phoenix/schemas/toolCallSchemas";
+import { assertUnreachable, isStringKeyedObject } from "@phoenix/typeUtils";
+import { safelyParseJSON } from "@phoenix/utils/jsonUtils";
+
+/**
+ * A partial tool call, the result of streaming a tool call back to the client
+ * The arguments are a string since they are not complete and thus not fully JSON serializable
+ */
+export type PartialOutputToolCall = {
+  id: string;
+  function: {
+    name: string;
+    arguments: string;
+  };
+};
+
+const isPartialOutputToolCall = (
+  toolCall: LlmProviderToolCall | PartialOutputToolCall
+): toolCall is PartialOutputToolCall => {
+  return (
+    typeof (toolCall as PartialOutputToolCall).function.arguments === "string"
```

Comment:
Is this safe? Particularly accessing `.function.arguments`

Reply:
yeah this was a quick fix last night, i'll clean this up to make it more safe, good call